crates/stages/src/stages/hashing_account.rs

```diff
@@ -190,27 +194,39 @@ impl<DB: Database> Stage<DB> for AccountHashingStage {
                 self.save_checkpoint(tx, checkpoint)?;
             }
 
+            let (sender, rx) = mpsc::unbounded_channel();
             let start_address = checkpoint.address.take();
             let next_address = {
                 let mut accounts = tx.cursor_read::<tables::PlainAccountState>()?;
 
-                let hashed_batch = accounts
-                    .walk(start_address)?
-                    .take(self.commit_threshold as usize)
-                    .map(|res| res.map(|(address, account)| (keccak256(address), account)))
-                    .collect::<Result<BTreeMap<_, _>, _>>()?;
```

Comment:
It is probably bcs of this. We collect hashed to BTreeMap and insert to the database in a sorted way.

There is probably some rayon helper that would iterate and transform values for us, and after the transformation is done, we sort it and insert it inside the db.